force-app/main/default/classes/service/SObjectFilterQueryService.cls

```diff
@@ -0,0 +1,164 @@
+public class SObjectFilterQueryService {
+    public static List<sObject> getFilteredRecords(String sObjectName, List<FilterConditionWrapper> filterConditions, String filterLogic, Integer recordSize){
+       
+       try{
+           //validate the inputs
+           if(String.isBlank(sObjectName)){
+             throw new IllegalArgumentException('SObject Name Cannot be Blank.');
+           }
+           if (filterConditions == null || filterConditions.isEmpty()) {
+            throw new IllegalArgumentException('Filter Conditions list cannot be empty.');
+        }
+        if (String.isBlank(filterLogic)) {
```

Comment:
Good checks. One thing, please pull out those lines of code into a separate method to divide the return early pattern and the domain logic.